core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -105,6 +107,10 @@ public InformationSchemaPageSource(
         table = tableHandle.getTable();
         prefixIterator = Suppliers.memoize(() -> {
             Set<QualifiedTablePrefix> prefixes = tableHandle.getPrefixes();
+            if (table == InformationSchemaTable.COLUMNS && prefixes.equals(defaultPrefixes("hive"))) {
```

Comment:
The prefix can't be hardcoded in the engine for two reasons:
1. The prefix is dynamic and based on configuration. It depends on how the administrator decided to install the hive plugin in the Trino cluster
2. The engine shouldn't have any specific knowledge about connectors. Connector-specific decisions should happen within connectors.